Detectors/TRD/base/src/ChamberStatus.cxx

```diff
@@ -0,0 +1,230 @@
+
+void ChamberStatus::setStatus(int det, char status)
+{
+
+  //
+  // set the chamber status
+  //
+  //
+
+  switch (status) {
+    case ChamberStatus::kGood:
+      mStatus[det] |= kGoodpat;
+      mStatus[det] &= !kNoDatapat;
+      mStatus[det] &= !kBadCalibratedpat;
+      break;
+    case ChamberStatus::kNoData:
+      mStatus[det] &= !kGoodpat;
+      mStatus[det] |= kNoDatapat;
+      mStatus[det] |= kNoDataHalfChamberSideApat;
+      mStatus[det] |= kNoDataHalfChamberSideBpat;
+      //      mStatus[det] |=  kBadCalibratedpat;
+      break;
+    case ChamberStatus::kNoDataHalfChamberSideA:
+      mStatus[det] |= kNoDataHalfChamberSideApat;
+      if (mStatus[det] & kNoDataHalfChamberSideB) {
+        mStatus[det] |= kNoDatapat;
+        mStatus[det] &= !kGoodpat;
+      }
+      break;
+    case ChamberStatus::kNoDataHalfChamberSideB:
+      mStatus[det] |= kNoDataHalfChamberSideBpat;
+      if (mStatus[det] & kNoDataHalfChamberSideApat) {
+        mStatus[det] &= !kGoodpat;
+        mStatus[det] |= kNoDatapat;
+      }
+      break;
+    case ChamberStatus::kBadCalibrated:
+      mStatus[det] &= !kGoodpat;
+      mStatus[det] |= kBadCalibratedpat;
+      break;
+    case ChamberStatus::kNotCalibrated:
+      mStatus[det] &= !kGoodpat;
+      mStatus[det] |= kNotCalibratedpat;
+      // mStatus[det] &= !kBadCalibratedpat;
+      break;
+    default:
+      mStatus[det] &= !kGoodpat;
+      mStatus[det] &= !kNoDatapat;
+      mStatus[det] &= !kNoDataHalfChamberSideApat;
+      mStatus[det] &= !kNoDataHalfChamberSideBpat;
+      mStatus[det] &= !kBadCalibratedpat;
+      mStatus[det] &= !kNotCalibratedpat;
+  }
+}
+//_____________________________________________________________________________
+void ChamberStatus::unsetStatusBit(int det, char status)
+{
+
+  //
+  // unset the chamber status bit
+  //
+  //
+
+  switch (status) {
+    case ChamberStatus::kGood:
+      mStatus[det] &= !kGoodpat;
+      break;
+    case ChamberStatus::kNoData:
+      mStatus[det] &= !kNoDatapat;
+      break;
+    case ChamberStatus::kNoDataHalfChamberSideA:
+      mStatus[det] &= !kNoDataHalfChamberSideApat;
+      break;
+    case ChamberStatus::kNoDataHalfChamberSideB:
+      mStatus[det] &= !kNoDataHalfChamberSideBpat;
+      break;
+    case ChamberStatus::kBadCalibrated:
+      mStatus[det] &= !kBadCalibratedpat;
+      break;
+    case ChamberStatus::kNotCalibrated:
+      mStatus[det] &= !kNotCalibratedpat;
+      break;
+    default:
+      mStatus[det] &= !(kGoodpat & kNoDatapat & kNoDataHalfChamberSideApat & kNoDataHalfChamberSideBpat & kBadCalibratedpat & kNotCalibratedpat);
+  }
+}
+//_____________________________________________________________________________
+TH2D* ChamberStatus::plot(int sm, int rphi)
+{
+  //
+  // Plot chamber status for supermodule and halfchamberside
+  // as a function of layer and stack
+  //
+
+  TH2D* h2 = new TH2D(Form("sm_%d_rphi_%d", sm, rphi), Form("sm_%d_rphi_%d", sm, rphi), 5, 0.0, 5.0, 6, 0.0, 6.0);
```

Comment:
Please do ``h2->SetDirectory(nullptr)`` to avoid segm.violation if the histo is created in some root file folder.

Reply:
will do